test/unit/services/awsAdapter_spec.js

```diff
@@ -65,8 +74,77 @@ describe("The AwsAdapter class ", function () {
 
 			var runInstancesStub;
 			var beginPollingStub;
+			var createSecurityGroupStub;
+			var authorizeSecurityGroupIngressStub;
 
 			before(function () {
+
+				createSecurityGroupStub = Sinon.stub(ec2, "createSecurityGroupAsync")
+				.resolves({ GroupId : DEFAULT_GROUP_ID });
+
+				authorizeSecurityGroupIngressStub = Sinon.stub(ec2, "authorizeSecurityGroupIngressAsync")
+				.resolves("test");
+
+				runInstancesStub = Sinon.stub(ec2, "runInstancesAsync").resolves({
+						Instances : [ {
+							InstanceId : "test"
+						} ]
+				});
+
+				beginPollingStub = Sinon.stub(awsAdapter,"beginPolling");
+
+				awsAdapter.runInstances(VALID_INSTANCE)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			after(function () {
+				runInstancesStub.restore();
+				beginPollingStub.restore();
+				createSecurityGroupStub.restore();
+				authorizeSecurityGroupIngressStub.restore();
+			});
+
+			it("creates the security group and returns a new instance with the ami and type provided", function () {
```

Comment:
It would be more helpful when reviewing tests to separate the validation of stub parameters into different `it` blocks. For instance, if the wrong parameters were passed to both the first and second stub, this `it` block will fail, but only print output for the first failure. After fixing the first stub, the block would fail again for the second. By separating `it` blocks to be specifically testing each stub, we can see all of the failures on the first pass and fix them more efficiently, without any hidden failures.